scrub_hub_vite/src/models/chat.ts

```diff
@@ -0,0 +1,197 @@
+import { fromBase64, toBase64 } from "../base64";
+import { ChatProps, MessageProps } from "../components/ChatComponent";
+import { ConversationListProps, ConversationProps } from "../components/ConversationListComponent";
+
+export type User = { name: string, id: number };
+function isUser(obj?: {[key: string]: unknown}): obj is User {
```

Comment:
Why is the value unknown? I don't like this very much LOL

Reply:
The value is obtained by parsing JSON. So it's unknown because we don't know what it is. `{[key: string]: unknown}` is a type that I believe every JavaScript object technically matches, and it makes it so TypeScript doesn't complain if I access properties of `obj`.
I first tried using type any, but eslint didn't like that. I'm not sure if using any would actually be appropriate here, but I decided to take their recommendation to not use any.

Do you have another idea of what it should be?

Reply:
Don't we know what should be in the JSON though? If we know what the JSON will contain we can make a type for the content of the JSON we are reading and thus I believe ensure type safety?? :)

Reply:
That is exactly what this code does! As I mentioned in another comment, I could move the JSON parsing into these functions to make it clearer what's happening. JSON string in, typed object out.

Reply:
YES PLEASE! If we keep just juggling with any and unknown types we might aswell just write PHP or Javascript code at that stage LMAO